### PR TITLE
feat: Avatar Group overflow spacing and overflow button border color

### PR DESCRIPTION
**Pull request by olshansky**
## Related Issue
Closes https://github.com/SAP/fundamental-styles/issues/2091
Part of https://github.com/SAP/fundamental-styles/issues/2153

## Description
Added overflow spacing and overflow button accent border. 

#### Please check whether the PR fulfills the following requirements

1. The output matches the design specs
- [x] All values are in `rem`
- [x] Text elements follow the truncation rules
- [x] hover state of the element follow design spec
- [x] focus state of the element follow design spec
- [x] active state of the element follow design spec
- [x] selected state of the element follow design spec
- [x] selected hover state of the element follow design spec
- [x] pressed state of the element follow design spec
- [x] Responsiveness rules - the component has modifier classes for all breakpoints
- [n/a] Includes Compact/Cosy/Tablet design
- [x] RTL support
2. The code follows fundamental-styles code standards and style
- [x] only one top level `fd-*` class is used in the file
- [x] BEM naming convention is used
- [x] Mixins are used for repeatable code (`fd-rtl`, `fd-ellipsis`, `fd-flex`, `fd-selected`, `fd-focus`, ect.)
- [x] A11y support - keyboard support, screenreader support, proper ARIA attributes, etc.
- [x] `fd-reset()` mixin is applied to all elements
- [x] Variables are used, if some value is used more than twice.
- [x] Checked if current components can be reused, instead of having new code.
3. Testing 
- [x] last commit message should have `[ci visual]` so it can trigger chromatic visual regression (e.g. `test: run chromatic visual regression [ci visual]`)
- [n/a] tested Storybook examples with "CSS Resources" `normalize` option 
- [n/a] tested Storybook examples with "CSS Resources" `unnormalize` option 
- [x] Verified all styles in IE11
- [x] Updated tests
4. Documentation
- [x] Storybook documentation has been created/updated
- [n/a] Breaking Changes [wiki](https://github.com/SAP/fundamental-styles/wiki/Breaking-Changes) has been updated in case of breaking changes.

### CODE REVIEW

File: src/avatar-group.scss [hunk 1]
```diff
@@ -81,6 +81,19 @@ $block: #{$fd-namespace}-avatar-group;
     }
   }
 
+  &__overflow {
+    @include fd-reset();
+
+    &__body {
```
Comment (mikerodonnell89) [inline src/avatar-group.scss:87]: results in `fd-avatar-group__overflow__body` which is not valid BEM, `fd-avatar-group__overflow-body` is fine

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]

File: src/avatar-group.scss [hunk 2]
```diff
@@ -81,6 +81,19 @@ $block: #{$fd-namespace}-avatar-group;
     }
   }
 
+  &__overflow {
```
Comment (InnaAtanasova) [inline src/avatar-group.scss:84]: why do you need this? It's used like this `<div class="fd-popover fd-avatar-group__overflow">` and it already has a reset coming from `fd-popover`. If you inspect the element, all of the rules are overwritten:
Screen Shot 2021-02-05 at 2 23 54 PM

You can only use `fd-avatar-group__overflow-body` where you apply the 1rem padding and use the specificity to add margin to the item.

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]

File: src/avatar-group.scss (hunk 1 above)
Comment (olshansky) [reply src/avatar-group.scss]: @mikerodonnell89 fixed

File: src/avatar-group.scss (hunk 2 above)
Comment (olshansky) [reply src/avatar-group.scss]: @InnaAtanasova fixed

File: src/avatar-group.scss [hunk 3]
```diff
@@ -81,6 +81,17 @@ $block: #{$fd-namespace}-avatar-group;
     }
   }
 
+  &__overflow--body {
```
Comment (InnaAtanasova) [inline src/avatar-group.scss:84]: should be `&__overflow-body`

Comment (InnaAtanasova) [reply src/avatar-group.scss:84]: `__overflow--body` means that there's a class `__overflow` and `--body` is modifying this class

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
High contrast colors for the "+8" in group mode are incorrect

Screen Shot 2021-02-10 at 11 20 57 AM
Screen Shot 2021-02-10 at 11 21 56 AM

Also, the margin-right in individual mode is correctly set to 0.125rem, but there is some additional space before the next avatar so I think they may still be spread too far apart
Screen Shot 2021-02-10 at 11 24 49 AM

**Comment by olshansky** [comment]
> High contrast colors for the "+8" in group mode are incorrect
> 
> Screen Shot 2021-02-10 at 11 20 57 AM
> Screen Shot 2021-02-10 at 11 21 56 AM

@mikerodonnell89 why it's incorrect? The button has a `color-accent-6` and as I see avatar has the same colors in HC mode 

> Also, the margin-right in individual mode is correctly set to 0.125rem, but there is some additional space before the next avatar so I think they may still be spread too far apart

fixed

File: src/avatar-group.scss (hunk 3 above)
Comment (olshansky) [reply src/avatar-group.scss]: @InnaAtanasova fixed

**Review by InnaAtanasova — APPROVED** [review]

**Comment by mikerodonnell89** [comment]
> @mikerodonnell89 why it's incorrect? The button has a `color-accent-6` and as I see avatar has the same colors in HC mode

@olshansky screenshot from the specs: 
Screen Shot 2021-02-16 at 6 36 16 PM

So I think we will need a new style for this overflow avatar with the + text

**Comment by olshansky** [comment]
> @olshansky screenshot from the specs:
> Screen Shot 2021-02-16 at 6 36 16 PM
> 
> So I think we will need a new style for this overflow avatar with the + text

@mikerodonnell89 screenshot from the examples (I just removed accent color class modifier). Looks correct, isn't it? 
FYI this button has a color different color accent in examples just to show that we could use color accents modifier classes for this button.
image

**Review by mikerodonnell89 — APPROVED** [review]

**Comment by InnaAtanasova** [comment]
Screen Shot 2021-02-18 at 3 22 37 PM


@olshansky are these changes intended?

**Comment by olshansky** [comment]
> @olshansky are these changes intended?

@InnaAtanasova old service for placeholder images don't work _(that's why I changed it at the [last commit](https://github.com/SAP/fundamental-styles/pull/2121/commits/77ec69df46c09c331b60317380f9296020ebea7c))_

**Comment by droshev** [comment]
@olshansky can you address https://github.com/SAP/fundamental-styles/issues/2153 as part of this ticket?

**Comment by olshansky** [comment]
@droshev should it be part of this PR? Because it is an issue in the popover component 
image

**Comment by droshev** [comment]
> @droshev should it be part of this PR? Because it is an issue in the popover component
> image

what is the problem with the popover?

**Comment by olshansky** [comment]
> what is the problem with the popover?

Popover header border color is different than the body border in Dark theme

**Comment by droshev** [comment]
> > what is the problem with the popover?
> 
> @droshev Popover header border color is different than the body border in Dark theme

my comment was related to the arrow in RTL - it wasn't centered

**Comment by olshansky** [comment]
> my comment was related to the arrow in RTL - it wasn't centered
> 
> https://user-images.githubusercontent.com/4380815/108005485-d86a8780-6fc6-11eb-9f0a-d0d91f6d454f.png

Oh yes, fixed

**Pull request merged**